lib/active_shipping/carriers/ups.rb

```diff
@@ -194,6 +195,14 @@ def get_delivery_date_estimates(origin, destination, packages, pickup_date=Date.
       parse_delivery_dates_response(origin, destination, packages, response, options)
     end
 
+    def void_shipment(tracking, options={})
+      options = @options.merge(options)
+      access_request = build_access_request
+      void_request = build_void_request(tracking)
+      response = commit(:void, save_request(access_request + void_request), (options[:test] || false))
```

Comment:
I also find the overloaded use of the `+` operator to be confusing. Is it concatenating two XML documents?